File: branches/Medusa/Source/Fantasy.Jobs/Choose.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.XSerialization;
using Fantasy.Jobs.Properties;
using System.Xml;

namespace Fantasy.Jobs
{
    [Instruction]
    [XSerializable("choose", NamespaceUri = Consts.XNamespaceURI)]  
    internal class Choose : AbstractInstruction
    {
        public override void Execute()
        {
            IJob job = (Job)this.Site.GetService(typeof(Job));
            IConditionService conditionService = (IConditionService)this.Site.GetService(typeof(IConditionService));
            When chose = null;
            int index = (int)job.RuntimeStatus.Local.GetValue("chose", -1);
            if (index == -1)
            {
                for (int i = 0; i < this.Cases.Count; i++)
                {
                    if (!String.IsNullOrWhiteSpace(this.Cases[i].Condition))
                    {
                        if (conditionService.Evaluate(this.Cases[i].Condition))
                        {
                            job.RuntimeStatus.Local["chose"] = i;
                            chose = this.Cases[i];
                            break;
                        }
                    }
                    else
                    {
                        throw new JobException(Properties.Resources.WhenRequireConditionText);
                    }
                }

                if (chose == null && this.Otherwise != null)
                {
                    job.RuntimeStatus.Local["chose"] = Int32.MaxValue;
                    chose = this.Otherwise;
                }

            }
            else if (index == Int32.MaxValue)
            {
                chose = this.Otherwise;
            }
            else
            {
                chose = this.Cases[index]; 
            }

            if (chose != null)
            {
                job.ExecuteInstruction(chose); 
            }

            
        }

        private IList<When> _cases = new List<When>();

        [XArray(Order=10)]
        [XArrayItem(Name="when", Type = typeof(When))]
        public IList<When> Cases
        {
            get { return _cases; }
        }

        [XElement("otherwise", Order=20)]
        public When Otherwise { get; set; }

#pragma warning disable 169
        [XNamespace]
        private XmlNamespaceManager _namespaces;
#pragma warning restore 169

    }


    
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/Choose.java
package fantasy.jobs;

import fantasy.xserialization.*;
import fantasy.jobs.Properties.*;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[Instruction, XSerializable("choose", NamespaceUri = Consts.XNamespaceURI)]
public class Choose extends AbstractInstruction
{
	@Override
	public void Execute()
	{
		IJob job = (IJob)this.getSite().getService(Job.class);
		IConditionService conditionService = (IConditionService)this.getSite().getService(IConditionService.class);
		When chose = null;
		int index = (int)job.getRuntimeStatus().getLocal().GetValue("chose", -1);
		if (index == -1)
		{
			for (int i = 0; i < this._cases.size(); i++)
			{
				if (!String.IsNullOrWhiteSpace(this._cases.get(i).Condition))
				{
					if (conditionService.Evaluate(this._cases.get(i).Condition))
					{
						job.getRuntimeStatus().getLocal().setItem("chose", i);
						chose = this._cases.get(i);
						break;
					}
				}
				else
				{
					throw new JobException(Properties.Resources.getWhenRequireConditionText());
				}
			}

			if (chose == null && this.Otherwise != null)
			{
				job.getRuntimeStatus().getLocal().setItem("chose", Integer.MAX_VALUE);
				chose = this.Otherwise;
			}

		}
		else if (index == Integer.MAX_VALUE)
		{
			chose = this.Otherwise;
		}
		else
		{
			chose = this._cases.get(index);
		}

		if (chose != null)
		{
			job.ExecuteInstruction(chose);
		}


	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XArray(Order = 10), XArrayItem(Name = "when", java.lang.Class = typeof(When))]
	private java.util.List<When> _cases = new java.util.ArrayList<When>();




//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XElement("otherwise", Order = 20)]
	public When Otherwise = null;

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
///#pragma warning disable 169
//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XNamespace]
	private XmlNamespaceManager _namespaces;
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
///#pragma warning restore 169

}